broker-daemon/block-order-worker/index.js

```diff
@@ -566,15 +566,23 @@ class BlockOrderWorker extends EventEmitter {
    * @returns {void}
    */
   async workLimitBlockOrder (blockOrder, targetDepth) {
-    if (blockOrder.timeInForce !== BlockOrder.TIME_RESTRICTIONS.GTC) {
-      throw new Error('Only Good-til-cancelled limit orders are currently supported.')
-    }
-
     const orderbook = this.orderbooks.get(blockOrder.marketName)
 
-    // fill as many orders at our price or better
+    // get available orders for the requested depth and price
     const { orders, depth: availableDepth } = await orderbook.getBestOrders({ side: blockOrder.inverseSide, depth: targetDepth.toString(), quantumPrice: blockOrder.quantumPrice })
-    await this._fillOrders(blockOrder, orders, targetDepth.toString())
+
+    if (blockOrder.timeInForce === BlockOrder.TIME_RESTRICTIONS.GTC) {
+      await this._fillOrders(blockOrder, orders, targetDepth.toString())
```

Comment:
[nit] might be nice to add some "why" here, similar to https://github.com/sparkswap/broker/pull/528/files#diff-a509eb0ce29028bbf0f9e9c2ecf846b5L575